kafka/producer/kafka.py

```diff
@@ -625,7 +625,7 @@ def _partition(self, topic, partition, key, value,
             assert partition in self._metadata.partitions_for_topic(topic), 'Unrecognized partition'
             return partition
 
-        all_partitions = list(self._metadata.partitions_for_topic(topic))
+        all_partitions = sorted(list(self._metadata.partitions_for_topic(topic)))
```

Comment:
sorted will operate on a set, so I think we can drop the explicit `list` conversion